test/functional/lua/net_spec.lua

```diff
@@ -39,6 +39,36 @@ describe('vim.net.request', function()
     )
   end)
 
+  it('detects filetype for remote content', function()
+    t.skip(skip_integ, 'NVIM_TEST_INTEG not set: skipping network integration test')
+
+    local ft = exec_lua([[
+      vim.cmd('runtime! plugin/nvim/net.lua')
+      vim.cmd('runtime! filetype.lua')
+      -- github raw dump of a small lua file in the neovim repo
+      vim.cmd('edit https://raw.githubusercontent.com/neovim/neovim/master/runtime/syntax/tutor.lua')
+      vim.wait(2000, function() return vim.bo.filetype ~= '' end)
+      return vim.bo.filetype
+    ]])
+
+    assert(ft == 'lua')
+  end)
+
+  it('removes the modified flag from the buffer for remote content', function()
+    t.skip(skip_integ, 'NVIM_TEST_INTEG not set: skipping network integration test')
+
+    local buffer_modified = exec_lua([[
+      vim.cmd('runtime! plugin/nvim/net.lua')
+      vim.cmd('edit https://raw.githubusercontent.com/neovim/neovim/master/runtime/syntax/tutor.lua')
+      -- wait for buffer to have content
+      vim.wait(2000, function() return vim.fn.wordcount().bytes > 0 end)
+      vim.wait(2000, function() return vim.bo.modified == false end)
+      return vim.bo.modified
+    ]])
+
+    assert(not buffer_modified)
```

Comment:
@tampueroc this is currently failing, but I'm not sure why, as when you open a remote url in neovim and then run `= vim.bo.modified` it returns false

Reply:
nevermind, fixed it by adding:

```
      vim.cmd('runtime! filetype.lua')
```